src/aiida_quantumespresso/workflows/ph/base.py

```diff
@@ -103,7 +102,23 @@ def get_builder_from_protocol(cls, code, parent_folder=None, protocol=None, over
         if 'settings' in inputs['ph']:
             builder.ph['settings'] = orm.Dict(inputs['ph']['settings'])
         builder.clean_workdir = orm.Bool(inputs['clean_workdir'])
-        builder.ph['qpoints'] = qpoints
+
+        if 'qpoints' in inputs['ph']:
+            qpoints_mesh = inputs['ph'].pop('qpoints')
+            qpoints = orm.KpointsData()
+            qpoints.set_kpoints_mesh(qpoints_mesh)
+            builder.ph['qpoints'] = qpoints
+        else:
+            inputs = {
+                'structure': structure,
+                'distance': orm.Float(inputs['qpoints_distance']),
```

Comment:
Does this actually work? I think the inputs are not guaranteed to have a `qpoints_distance` do they? I think a default value should be added to the protocol file.

Reply:
Yes, I think it would be the best way to add `qpoints_distance` in the protocol, and also to create a dedicated input in the workchain. Maybe to do a implementation similar that kpoints in `PwBaseWorkChain`. However i dont know if I should also add qpoints in PhWorkChain

Reply:
Ah, I now see this comment. This is exactly what I described in my response to the other comment, so yes, I think this should be the correct approach.